In [ ]:
# First Exploration – Sentiment Scoring Test
#Load a dictionary-based sentiment analyzer and try it on a few example headlines.

#use VADER
#VADER is a pre-built sentiment analysis tool optimized for social media and news-like short text — it’s great for quick prototyping.
#It captures emphasis through things like capitalization, punctuation, and even emojis, which helps in informal or headline-style text.
#Since VADER relies on a fixed lexicon, it can misread complex headlines

In [1]:
# Install VADER if needed
%pip install vaderSentiment

Defaulting to user installation because normal site-packages is not writeable

[notice] A new release of pip is available: 24.3.1 -> 25.1.1
[notice] To update, run: /Applications/Xcode.app/Contents/Developer/usr/bin/python3 -m pip install --upgrade pip


In [2]:

from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

analyzer = SentimentIntensityAnalyzer()

# Sample headline
headline = "Markets soar as inflation fears ease"

# Get sentiment score
score = analyzer.polarity_scores(headline)
print(score)


{'neg': 0.301, 'neu': 0.43, 'pos': 0.269, 'compound': -0.0772}


Observations: 

30.1% of the sentence has negative tone.

43% is neutral content.

26.9% is positive.

compound = -0.0772 - Final overall sentiment is slightly negative.